### ISSUE TITLE
Implement Performance Insight screen

### ISSUE THREAD

**Issue by akul43** [CSU-Northridge-ARCS-Dev/smart-clothing-app#51]
<img width="223" alt="image" src="https://github.com/CSU-Northridge-ARCS-Dev/smart-clothing-app/assets/46544707/35cace99-ae24-45e3-acf8-9137b6fcefac">

**Comment by akul43** [comment]
Standby

**Comment by qtdinh** [comment]
**Basic outline of the implementation:**
- Create the front-end for the interface based on [Figma.](https://www.figma.com/file/2rLsjlDrHHVE4YAbP65Mny/Updated-Smart-Clothing-Mobile-Wireframe?type=design&node-id=0-1&mode=design&t=IYJJSBjGX2Uw1MP7-0)
- Look into Activity Rings, and create a component out of it.
- Look into the bar graphs, and create a component out of it.
- Look into displaying data based on hard-coded values.
- Possible Redux to leverage the component in different screens and daily metrics to fetch and carry over data.

**Doling out tasks:**
- Matthew
  - Responsible for displaying the date and icons on the top of the screen.
  - Responsible for creating the bar graphs and displaying data on them.
- Thomas
  - Responsible for creating the rings component and displaying data on them.
  - Responsible for daily insights.
  
**Resources:**
[Activity Rings library](https://github.com/thecodehunter/react-native-activity-rings)
[Activity Rings tutorial](https://www.youtube.com/watch?v=g5O2GJ1r9D4)

**Comment by qtdinh** [comment]
I'm beginning to think with the project at its current state, a perfect Activity Ring may not be feasible.

**Some problems I've faced:**
- Trying to bring in the various packages that work with all the different libraries while doing it in JavaScript is a nightmare.
- SVG circle stroke does not let it go past 100%. The way it functions is it does a full revolution, from which it starts to undo the ring until it reaches 200%. The same pattern continues for any following percentages.
- The components that would be perfect for our purposes are in TypeScript. I tried to get one working, but I could not wrap my head around it. Another up-to-date component that I tried involves a custom package/library that works with Reanimated 3, but considering we use Expo.

**In summary, there's not a lot of support for React Native regarding Activity Rings the way we're doing it right now. I've ran into so many problems, and I'm not sure if we can do the circular overflow effect or not. 

Such a thing would either require a lot more time to figure out how to make it work, but I'm not sure if I'm capable of that.**

**Comment by qtdinh** [comment]
![image](https://github.com/CSU-Northridge-ARCS-Dev/smart-clothing-app/assets/59325099/2a6cc785-748d-42c1-a113-4bdc548f3c38)

Scratch that, might be able to get it working! Using the React Native Skia library. :)

**Comment by qtdinh** [comment]
![image](https://github.com/CSU-Northridge-ARCS-Dev/smart-clothing-app/assets/59325099/3623af26-3e47-45a7-8440-e139f79393b4)

Done with my part of the task. Will help Matthew out to push this to completion.

**Comment by qtdinh** [comment]
[ringsnbars.webm](https://github.com/CSU-Northridge-ARCS-Dev/smart-clothing-app/assets/59325099/405a28ea-fb7f-462b-b37c-b85d52bb9097)

**Done with the renders of the bar graphs.**
- Leveraged Victory Native XL, which utilizes Reanimated 3 and the Skia library to create more highly performant data visualizations along with options for animations and gestures.
- Created an Activity Chart component that we can reuse for the bar charts.
- Researched why the [Activity Rings crash](https://github.com/CSU-Northridge-ARCS-Dev/smart-clothing-app/issues/70) and identified the root of the problem.

**To-do**
- Make sure graphs work in conjunction with the rings, e.g. 800 is max capacity, one of the rings needs to grab the percentage accordingly.
- Thinking of creating a gesture handler where the user can click and drag over the bar chart to view calories burned for each tick like a slider: [see example.](https://formidable.com/open-source/victory-native/)
- Still need to talk to the back-end team to discuss data processing.
